saleor/graphql/order/mutations/draft_orders.py

```diff
@@ -99,16 +97,62 @@ class Meta:
         error_type_field = "order_errors"
 
     @classmethod
-    def clean_channel_id(cls, instance, channel_id):
-        if channel_id and hasattr(instance, "channel"):
-            raise ValidationError(
-                {
-                    "channel": ValidationError(
-                        "Can't update existing order channel id.",
-                        code=OrderErrorCode.NOT_EDITABLE.value,
-                    )
-                }
-            )
+    def clean_input(cls, info, instance, data):
+        shipping_address = data.pop("shipping_address", None)
+        billing_address = data.pop("billing_address", None)
+        redirect_url = data.pop("redirect_url", None)
+
+        cleaned_input = super().clean_input(info, instance, data)
+
+        channel_id = data.get("channel_id", None)
+        if "channel_id" in cleaned_input and channel_id is None:
```

Comment:
Maybe I am missing something but instead of removing channel_id from cleaned_data you could `.pop` it before clean_input call, right? You already have method dedicated for cleaning channel, so not sure if we need to pass it to clean_input. Something like this:
```
  @classmethod
 def clean_input(cls, info, instance, data):
     shipping_address = data.pop("shipping_address", None)
     billing_address = data.pop("billing_address", None)
     redirect_url = data.pop("redirect_url", None)
     channel_id = data.pop("channel_id", None)

     cleaned_input = super().clean_input(info, instance, data)

     channel = cls.clean_channel_id(info, instance, cleaned_input, channel_id)
```